### PR TITLE
Parcel ut partial front end 

### PR DESCRIPTION
**Pull request by iaincrawf**
Have kept it out of the menu for now

Search parcels is implemented from front end, still need to finish full query in service as currently on searching on parcel number

Still need to do add/edit parcel page

### CODE REVIEW

File: src/Service.Host/client/src/components/parcels/ParcelsSearch.js [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+                <Grid item xs={2}>
+                    <SearchInputField
+                        label="Date Created"
+                        fullWidth
+                        placeholder="search.."
+                        onChange={handleDateSearchChange}
+                        propertyName="dateCreatedSearchTerm"
+                        type="date"
+                        value={dateCreatedSearch}
+                        debounce={5000}
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/parcels/ParcelsSearch.js:235]: long debounce?

File: src/Service.Host/client/src/components/parcels/ParcelsSearch.js [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        setSearchTerm(
+            `${parcelNumberSearch}&supplierIdSearchTerm=${supplierIdSearch}&carrierIdSearchTerm=${carrierIdSearch}&dateCreatedSearchTerm=${dateCreatedSearch}&supplierInvNoSearchTerm=${supplierInvNoSearch}&consignmentNoSearchTerm=${args[1]}&commentsSearchTerm=${commentsSearch}`
+        );
+    };
+
+    const handleCommentSearchChange = (...args) => {
+        setCommentsSearch(args[1]);
+        setSearchTerm(
+            `${parcelNumberSearch}&supplierIdSearchTerm=${supplierIdSearch}&carrierIdSearchTerm=${carrierIdSearch}&dateCreatedSearchTerm=${dateCreatedSearch}&supplierInvNoSearchTerm=${supplierInvNoSearch}&consignmentNoSearchTerm=${consignmentNoSearch}&commentsSearchTerm=${args[1]}`
+        );
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/parcels/ParcelsSearch.js:93]: These search strings are quite unwieldly but I'm not sure there's a better option..
You could store all the searchTerms in one state object

```
const [searchTerms, setSearchTerms] = useState({
    parcelNumberSearch: '',
    supplierIdSearch: '',
   etc... }
);
```

and then write a little function to build the big query strings by looping over the searchTerms keys and values, or something like that.

If it ends up being even messier then just leave it, but a little utility function (or hook?) to turn objects to searchTerms could be useful

Comment (lewisrenfrew) [reply src/Service.Host/client/src/components/parcels/ParcelsSearch.js:93]: we also already have this package as a dependency that could help https://www.npmjs.com/package/query-string

**Review by lewisrenfrew — COMMENTED** [review]
looks good. few commented out sections you could remove just for cleanliness but I imagine you'll be coming back to this soon anyway

File: src/Service.Host/client/src/components/parcels/ParcelsSearch.js (hunk 2 above)
Comment (richardiphillips) [reply src/Service.Host/client/src/components/parcels/ParcelsSearch.js:93]: +1 query-string

File: src/Service.Host/client/src/components/parcels/ParcelsSearch.js (hunk 1 above)
Comment (iaincrawf) [reply src/Service.Host/client/src/components/parcels/ParcelsSearch.js:235]: I don't think these even take a debounce param, will take it out

File: src/Service.Host/client/src/components/parcels/ParcelsSearch.js (hunk 2 above)
Comment (iaincrawf) [reply src/Service.Host/client/src/components/parcels/ParcelsSearch.js:93]: query string sounds good to me, I'll look it up properly and put it in

**Pull request closed without merging**